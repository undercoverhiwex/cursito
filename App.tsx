/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React, {useState} from 'react';
import {
  SafeAreaView,
  StatusBar,
  StyleSheet,
  Text,
  TouchableOpacity,
  useColorScheme,
} from 'react-native';

import {Colors} from 'react-native/Libraries/NewAppScreen';

import Icon from 'react-native-vector-icons/FontAwesome';
import {Search} from './components/Search';
import api from './http/fetch';

function App(): React.JSX.Element {
  const [counter, setCounter] = useState(0);
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  const handlePress = () => setCounter(curr => curr + 1);

  return (
    <SafeAreaView style={[backgroundStyle, styles.container]}>
      <StatusBar
        barStyle={isDarkMode ? 'light-content' : 'dark-content'}
        backgroundColor={backgroundStyle.backgroundColor}
      />

      <Text style={styles.text} testID="counter-label">
        {counter}
      </Text>

      <TouchableOpacity
        style={styles.button}
        onPress={handlePress}
        testID="counter-button">
        <Icon name="plus" size={30} color="#dfe6e9" />
      </TouchableOpacity>

      <Search onPress={api.get} />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    fontSize: 30,
    marginBottom: 30,
  },
  button: {
    width: '90%',
    height: 45,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#ff7675',
    borderRadius: 20,
    marginBottom: 30,
  },
  input: {
    width: '90%',
    height: 45,
    backgroundColor: '#dfe6e9',
    borderRadius: 20,
    borderColor: '#000000',
    paddingHorizontal: 20,
  },
});

export default App;
