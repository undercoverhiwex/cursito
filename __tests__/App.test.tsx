/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: import explicitly to use the types shipped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import {fireEvent, render, screen} from '@testing-library/react-native';

describe('check render', () => {
  it('renders correctly', () => {
    render(<App />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('renders correctly counter and button', () => {
    const page = render(<App />);
    expect(page.getByTestId('counter-label')).toBeTruthy();
    expect(page.getByTestId('counter-button')).toBeTruthy();
  });
});

describe('check counter', () => {
  it('check correclty counter', () => {
    const page = render(<App />);
    const button = page.getByTestId('counter-button');
    fireEvent.press(button);
    expect(page.getByText('11')).toBeTruthy();
    fireEvent.press(button);
    expect(page.getByText('12')).toBeTruthy();
  });
});
