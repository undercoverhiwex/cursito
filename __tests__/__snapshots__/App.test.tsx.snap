// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render renders correctly 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "backgroundColor": "#F3F3F3",
      },
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      {
        "fontSize": 30,
        "marginBottom": 30,
      }
    }
    testID="counter-label"
  >
    0
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#ff7675",
        "borderRadius": 20,
        "height": 45,
        "justifyContent": "center",
        "marginBottom": 30,
        "opacity": 1,
        "width": "90%",
      }
    }
    testID="counter-button"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#dfe6e9",
            "fontSize": 30,
          },
          undefined,
          {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "100%",
      }
    }
    testID="search-component"
  >
    <TextInput
      onChangeText={[Function]}
      style={
        {
          "backgroundColor": "#dfe6e9",
          "borderColor": "#000000",
          "borderRadius": 20,
          "height": 45,
          "marginBottom": 30,
          "paddingHorizontal": 20,
          "width": "90%",
        }
      }
      testID="search-input"
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#0984e3",
          "borderRadius": 20,
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 30,
          "opacity": 1,
          "width": "90%",
        }
      }
      testID="search-button"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#dfe6e9",
              "fontSize": 30,
            },
            undefined,
            {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
